Mathlib/Order/Monotone/Basic.lean

```diff
@@ -1068,6 +1068,20 @@ theorem Monotone.prod_map (hf : Monotone f) (hg : Monotone g) : Monotone (Prod.m
 theorem Antitone.prod_map (hf : Antitone f) (hg : Antitone g) : Antitone (Prod.map f g) :=
   fun _ _ h ↦ ⟨hf h.1, hg h.2⟩
 
+lemma monotone_separately_iff_jointly {h : α × β → γ} :
+    (∀ a, Monotone (h ∘ Prod.mk a)) ∧ (∀ b, Monotone (fun a => h (a,b))) ↔ Monotone h :=
+  ⟨fun ⟨h₂, h₁⟩ _ _ hab =>
+      le_trans (h₁ _ (Prod.mk_le_mk.mp hab).1) (h₂ _ (Prod.mk_le_mk.mp hab).2),
+    fun h => ⟨fun _ _ _ hab => h (Prod.mk_le_mk_iff_right.mpr hab),
+      fun _ _ _ hab => h (Prod.mk_le_mk_iff_left.mpr hab)⟩ ⟩
```

Comment:
I think this will be more readable with
```lean4
... where
  mp ⟨h₂, h₁⟩ _ _ hab := ...
  mpr h := ...
```
and also below

Reply:
Done, thanks.